### PR TITLE
Astro: Added sun phase

### PR DESCRIPTION
**Pull request by gerrieg**
a wish from https://community.openhab.org/t/sun-raised-set-indicator/12834/24

### CODE REVIEW

File: addons/binding/org.openhab.binding.astro/ESH-INF/thing/channels.xml [hunk 1]
```diff
@@ -215,7 +215,37 @@
...
+		<description>Sunphases</description>
+		<channels>
+			<channel id="name" typeId="sunPhaseName" />
+		</channels>
+	</channel-group-type>
+
+	<channel-type id="sunPhaseName">
+		<item-type>String</item-type>
+		<label>Sunphase name</label>
+		<description>The name of the current sun phase</description>
```
Comment (ThomDietrich) [inline addons/binding/org.openhab.binding.astro/ESH-INF/thing/channels.xml]: "The current sun phase"

File: addons/binding/org.openhab.binding.astro/README.md [hunk 2]
```diff
@@ -32,6 +32,8 @@ Optionally, a refresh interval (in seconds) can be defined to also calculate pos
         * **channel**: `spring, summer, autumn, winter` (DateTime), `name` (String)
     * **group** `eclipse`
         * **channel**: `total, partial, ring` (DateTime)
+    * **group** `phase`
+        * **channel**: `name` (String)
```
Comment (ThomDietrich) [inline addons/binding/org.openhab.binding.astro/README.md]: Shouldn't the possible values be documented here (also for other channels)

**Comment by gerrieg** [comment]
@ThomDietrich 
Updated as suggested

File: addons/binding/org.openhab.binding.astro/README.md (hunk 2 above)
Comment (rtvb) [inline addons/binding/org.openhab.binding.astro/README.md]: Yes, please. And please also add a `sunphase.map` file to the full example.

**Comment by hakan42** [comment]
The SunCalc.java and MoonCalc.java classes seem to have a major change in tabs vs. spaces. Could you check that again please?

**Comment by kaikreuzer** [comment]
> The SunCalc.java and MoonCalc.java classes seem to have a major change in tabs vs. spaces. Could you check that again please?

It seems he only applied the code formatter on them. At least the new version is the correctly formatted one, so all looks good to me.

**Comment by ThomDietrich** [comment]
@kaikreuzer Maybe it would be meaningful to provide `org.eclipse.cdt.core.formatter` settings, including the "Format on Save" action, with this project? I see this happening a lot and the above is taking care of that perfectly in another project of mine.

**Comment by kaikreuzer** [comment]
@ThomDietrich The "format on save" is already in place. The code that was changed in this PR was most likely last edited when the formatting was not yet enforced.

**Pull request merged**